src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Automation/VsProjectEvents.cs

```diff
@@ -6,29 +6,32 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
 {
-    [Export(typeof(VSProjectEvents))]
-    [AppliesTo(ProjectCapability.VisualBasic)]
+    [Export(typeof(VSLangProj.VSProjectEvents))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
     [Order(Order.Default)]
-    internal class VisualBasicVSProjectEvents : VSProjectEvents
+    internal class VSProjectEvents : VSLangProj.VSProjectEvents
     {
         private readonly VSLangProj.VSProject _vsProject;
+        private readonly BuildManager _buildManager;
 
         [ImportingConstructor]
-        public VisualBasicVSProjectEvents(
+        public VSProjectEvents(
             [Import(ExportContractNames.VsTypes.CpsVSProject)] VSLangProj.VSProject vsProject,
-            UnconfiguredProject project)
+            UnconfiguredProject project,
+            BuildManager buildManager)
         {
             _vsProject = vsProject;
             ImportsEventsImpl = new OrderPrecedenceImportCollection<ImportsEvents>(projectCapabilityCheckProvider: project);
+            _buildManager = buildManager;
         }
 
         [ImportMany]
         protected OrderPrecedenceImportCollection<ImportsEvents> ImportsEventsImpl { get; set; }
 
         public ReferencesEvents ReferencesEvents => _vsProject.Events.ReferencesEvents;
 
-        public BuildManagerEvents BuildManagerEvents => _vsProject.Events.BuildManagerEvents;
+        public BuildManagerEvents BuildManagerEvents => _buildManager as BuildManagerEvents ?? _vsProject.Events.BuildManagerEvents;
 
-        public ImportsEvents ImportsEvents => ImportsEventsImpl.First().Value;
+        public ImportsEvents ImportsEvents => ImportsEventsImpl.FirstOrDefault()?.Value ?? _vsProject.Events.ImportsEvents;
```

Comment:
What does C# legacy return for this property? ImportsEvent is for VB only.

Reply:
Ah, I thought this started returning null, so it's not a change in behavior, and hence not changed in this PR. Legacy is relevant because we need to be compatible with it.

Reply:
I can actually revert all of the changes to this file as it turns out. CPS's OAVSProjectEvents gets its BuildManagerEvents through a VSProject import, which ends up being our implementation anyway.